pinot-spi/src/main/java/org/apache/pinot/spi/utils/LoopUtils.java

```diff
@@ -29,10 +28,9 @@ private LoopUtils() {
   public static final int MAX_ENTRIES_KEYS_MERGED_PER_INTERRUPTION_CHECK_MASK = 0b1_1111_1111_1111;
 
   // Check for thread interruption, every time after merging 8192 keys
-  public static void checkMergePhaseInterruption(int mergedKeys) {
-    if ((mergedKeys & MAX_ENTRIES_KEYS_MERGED_PER_INTERRUPTION_CHECK_MASK) == 0
-        && Tracing.ThreadAccountantOps.isInterrupted()) {
-      throw new EarlyTerminationException("Interrupted while merging records");
+  public static void sampleAndCheckInterruptionPeriodically(int mergedKeys) {
```

Comment:
my suggestion would be to keep this in `ThreadAccountantOps` similar to `sampleAndCheckInterruption()` method to keep all related code together unless you see this particular class evolving into something different ?

Reply:
yes moved